### ISSUE TITLE
Diff testing: iteratively improve

### ISSUE THREAD

**Issue by danwt** [cosmos/interchain-security#136]
The commit [_eb4af0d1b90959bfa8a8882ebc85fe5a65fc8f43_ in ](https://github.com/cosmos/interchain-security/tree/eb4af0d1b90959bfa8a8882ebc85fe5a65fc8f43) on the [PR for differential testing](https://github.com/cosmos/interchain-security/pull/126) is a milestone. Meaningful further work on the differential testing is now blocked by

- https://github.com/cosmos/interchain-security/issues/39

Essentially, the PR implements TDD and the current SUT is not sufficient to make the tests pass.

Many tasks are needed for future work including

- [ ] https://github.com/cosmos/interchain-security/issues/137
- [ ] https://github.com/cosmos/interchain-security/issues/132
- [ ] https://github.com/cosmos/interchain-security/issues/133
- [ ] https://github.com/cosmos/interchain-security/issues/134
- [ ] https://github.com/cosmos/interchain-security/issues/127
- [ ] https://github.com/cosmos/interchain-security/issues/135
- [x] https://github.com/cosmos/interchain-security/issues/140
